### ISSUE TITLE
Accessibility improvements

### ISSUE THREAD

**Issue by amcdnl** [reaviz/reaviz#310]
### Discussed in https://github.com/reaviz/reaviz/discussions/309

<div type='discussions-op-text'>

<sup>Originally posted by **maggiewachs** October 24, 2025</sup>
Hello! Our team recently started using Reaviz for data visualizations, and we've been impressed with how well it works with keyboard and screen readers just out of the box. We work for a US federal agency on websites like [this one](https://data.cms.gov/provider-data/) that support data transparency as part of the larger government open data initiative, so web accessibility is very important to us — it's my primary role on our team.

I wanted to see if your team is open to feedback and possible contributions that would improve the accessibility of your charts? (We're using the bar and pie charts, so that's where we'd start.) We've identified changes that would make them conform to the [Web Content Accessibility Guidelines (WCAG) 2.2](https://www.w3.org/TR/WCAG22/), the standard for measuring accessibility in web sites. There are very few open source dataviz tools that currently do this well (or at all), so it would be a real win. 

Future improvements we'd recommend and would be happy to chat about more: 
* Consider using background patterns to distinguish chart segments from each other, in addition to color. It's really, really difficult to find a color palette where each color is so distinct it holds up to all forms of low vision and colorblindness tests. Using background patterns (stripes in varying directions, crosshatches, dots, etc), icons, and a variety of text labels can do that work.
* Make tooltips that appear on hover accessible to keyboard users; mouse over/out events only work with a cursor. Each bar and pie piece is already keyboard focusable (hooray), so could this be a low effort fix?
* Style the default focus indicator to be more distinct against the element, so when keyboard users are navigating through the bar chart, the "you are here" feedback is clearer. It would also be super neat to see a focus indicator that conforms to the shape of each pie piece (right now it's a rectangle that kinda-sorta outlines the piece).
* Improve [focus order](https://www.w3.org/WAI/WCAG22/Understanding/focus-order.html) by reordering elements in the source code. We can use the Tab key to navigate the bar chart (awesome, we love that), but the order is backwards — users start from the bottom of the chart and work up, when it should be top-down.
* Support link semantics in axis and legend label markup for screen reader users. If something looks like a link, it must be marked up with semantic HTML (button, a) or attributes and event bindings that communicate the same meaning (ARIA roles, tabindex + JavaScript keyup event), otherwise a screen reader user can't engage the link. 
* Each chart's focusable SVG needs an accessible label. Ideally we could just associate the SVG with the visible heading (h3) using [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-labelledby), but using [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-label) would be a solid default.

...that was a lot, so I'm going to stop here and look forward to any replies. Thanks!</div>

**Comment by vishakha1801** [comment]
Hi! I noticed the recent commit addressed several of the accessibility items. Are any of the remaining sub-tasks still open for contribution? I would love to pick one up if so

**Comment by amcdnl** [comment]
@vishakha1801 - Yes, I did a basic pass but need to test thoroughly it to ensure everything is good. You are welcome to pick up anything additional tho!

**Comment by maggiewachs** [comment]
Hey folks! I was wondering if this feature could be taken on next? @vishakha1801, if you're interested? 

> Each chart's focusable SVG needs an accessible label. Ideally we could just associate the SVG with the visible heading (h3) using [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-labelledby), but using [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-label) would be a solid default.

**Comment by amcdnl** [comment]
Go for it! I haven't had time lately to pick these up but happy to provide feedback/etc.
